hw/ip/aes/doc/aes.md

```diff
@@ -0,0 +1,296 @@
+For AES-128 and AES-192, the actual initial key used for encryption is formed by taking the values in Initial Key register 0-3 and 0-5, respectively.
+
+
+{{% section2 Block Operation }}
+
+For block operation, software must initialize the AES unit as described in the previous section and then:
+1. Make sure that the AES unit:
+   1. Automatically starts encryption/decryption when new input data indicated by the MANUAL_START_TRIGGER bit in the Control register being 1'b0.
+   2. Does not overwrite previous output data that has not been read by the processor indicated by the FORCE_DATA_OVERWRITE bit in the Control register being 1'b0.
+2. Write Input Data Block 0 to the Input Data register.
+3. Wait for the INPUT_READY bit in the Status register to become 1'b1, i.e. wait for the AES unit to load Input Data Block 0 into the internal State register and start operation.
+4. Write Input Data Block 1 to the Input Data register.
+
+Then for every Data Block I=0,..N-1, software must:
+1. Wait for the OUTPUT_VALID bit in the Status register to become 1'b1, i.e., wait for the AES unit to finish encryption/decryption of Block I.
+2. Read Output Data Block I from the Output Data register.
+3. Write Input Data Block I+1 into the Input Data register.
+
+For the last Data Block N, software must:
+1. Wait for the OUTPUT_VALID bit in the Status register to become 1'b1, i.e., wait for the AES unit to finish encryption/decryption of Block N.
+2. Read Output Data Block N from the Output Data register.
+
+
+{{% section2 De-Initialization}}
+
+After finishing operation, software must:
+1. Disable the AES unit to no longer automatically start encryption/decryption by setting the MANUAL_START_TRIGGER bit in the Control register to 1'b1.
+2. Clear the configured initial key by overwriting the Initial Key register.
+3. Clear the previous input data by overwriting the Input Data register.
+4. Clear the internal key registers and the Output Data register by setting the KEY_CLEAR and DATA_OUT_CLEAR bits in the TRIGGER register to 1'b1.
+
+
+{{% section2 Register Table }}
+
+The AES unit uses 8 and 4 separate write-only registers for the initial key and input data, as well as 4 separate read-only registers for the output data.
+Compared to first-in, first-out (FIFO) interfaces, having separate registers has a couple of advantages:
+- Supported out-of-the-box by the register tool (the FIFO would have to be implemented separately).
+- Usability: critical corner cases where software updates input data or the key partially only are easier to avoid using separate registers and the `hwqe`-signals provided by the Register Tool.
+- Easier interaction with DMA engines
+Also, using a FIFO interface for something that is not actually FIFO (internally, 16B of input/output data are consumed/produced at once) is less natural.
+
+For a detailed overview of the register tool, please refer to the [Register Tool documentation.](../../../../doc/rm/register_tool.md)
```

Comment:
I assume this has already been converted to `<relref>` in subsequent PR?

Reply:
Yes, that's converted.